src/dotnet/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -63,17 +63,13 @@ private async Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(Location
 
         using (var activity = ActivitySource.StartActivity())
         {
-            BalancingAuthority? balancingAuthority;
+            BalancingAuthority balancingAuthority;
             try
             {
-                var geolocation = this.LocationSource.ToGeopositionLocation(location);
+                var geolocation = await this.LocationSource.ToGeopositionLocationAsync(location);
                 balancingAuthority = await WattTimeClient.GetBalancingAuthorityAsync(geolocation.Latitude.ToString() ?? "", geolocation.Longitude.ToString() ?? "");
-
-                if(balancingAuthority == null) {
-                    throw new LocationConversionException();
-                }
             }
-            catch(LocationConversionException ex)
+            catch(Exception ex) when (ex is LocationConversionException ||  ex is WattTimeClientException)
```

Comment:
Might be better to explicitly catch these two types of exceptions in their own exception catch block (see section here on multiple blocks for an example: https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/exceptions/exception-handling)

Reply:
I did think about that. But in this case, the handling for both the exceptions is the same. So decided to keep it together instead of duplicating the code